### PR TITLE
Refactor assoc-to-fn assoc-to-fn-unbalanced.

### PR DESCRIPTION
**Pull request by zhouhengmin**
- src/clj-template/core.clj: use auto-gensym in syntax-quote to simplify the code.

### CODE REVIEW

File: src/clj_template/core.clj [hunk 1]
```diff
@@ -38,10 +47,12 @@
...
-                (if (and (map? ~(symbol "attrs")) (not (empty? ~(symbol "attrs"))))
-                  (str " " (clojure.string/join " " (for [[~(symbol "k") ~(symbol "v")] ~(symbol "attrs")]
-                                                       (str (name ~(symbol "k")) "=\"" ~(symbol "v") "\""))) " />") " />")]
-            (str "<" ~markup-tag ~(symbol "attrs"))))))
+       `(defn ~(symbol (str markup-tag "-"))
+          [& [attrs# & body#]]
+          (let [str-attrs#
+                (if (and (map? attrs#)
+                         (not (empty? attrs#)))
+                  (str " " (gen-attributes attrs#) "/>")
```
Comment (brennonyork) [inline src/clj_template/core.clj]: this line should include `" />"` (note the space)

**Comment by brennonyork** [comment]
Thanks for the PR! Just one minor nit, if you could fix that I'd be happy to pull this in :)

**Comment by zhouhengmin** [comment]
OK, I fixed the missed space. :)

**Comment by zhouhengmin** [comment]
hi, brennonyork, is there any other problem with the pull request?

**Comment by brennonyork** [comment]
sorry, nope all good. Thanks!

**Pull request merged**